cornerstonejs/cornerstone3D
# feat(adapter): add RTSS Adapter and Labelmaps to Contours convertor

-extracted RTSS logic extension from OHIF viewer PR 3092







### Context
-extracted RTSS logic extension from OHIF viewer PR 3092



### Changes & Results



generateROIContoursFromSegmentations:
-use vtkImageMarchingSquares to iterate through segmentation labelmap and create contours that can be stored as ROIContours
RTSS:
-populate data structure compliant in 

Note: vtkjs v27.3.1 dependency added to adapters
### Testing


Sample usage of adapter under:
https://github.com/dxlin/viewers/tree/feat/RTSSadapter
->extensions/dicom-rtss/src/commandsModule.js

Using viewer:
1. Start viewer
2. Select Water Phantom study (or any with CT and PT) in TMTV mode
3. Create "New Label" on right right
4. Select "Rectangle ROI Threshold Tool"
5. Create arbitrary rectangle in axial PT viewport
6. Click "Run"
7. Click "Export RTSS"

Save prompt should show up to save RTSS dicom file. Can also view console to review contours and RTSS object generated.
### Checklist

#### PR



- [] My Pull Request title is descriptive, accurate and follows the
  semantic-release format and guidelines.

#### Code

- [] My code has been well-documented (function documentation, inline comments,
  etc.)

- [] I have run the `yarn build:update-api` to update the API documentation, and have
  committed the changes to this PR. (Read more here https://www.cornerstonejs.org/docs/contribute/update-api)


#### Public Documentation Updates



- [] The documentation page has been updated as necessary for any public API
  additions or removals.

#### Tested Environment

- [] "OS:  Windows 10
- [] "Node version:  18.16.0
- [] "Browser:
   Edge v115.0.1901.188


[blog]: https://circleci.com/blog/triggering-trusted-ci-jobs-on-untrusted-forks/
[script]: https://github.com/jklukas/git-push-fork-to-upstream-branch
